scripts/quests/outlands/Like_Shining_Leggings.lua

```diff
@@ -0,0 +1,108 @@
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return (status == QUEST_AVAILABLE and player:getFameLevel(xi.quest.fame_area.NORG) >= 3)
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heizo'] = quest:progressEvent(127),
+
+            onEventFinish =
+            {
+                [127] = function(player, csid, option, npc)
+                    quest:begin(player) -- Start Like Shining Leggings
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heizo'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(128, player:getCharVar(leggingsVarName))-- Update the player on the number of Rusty Leggings they have turned in
+                end,
+
+                onTrade = function(player, npc, trade)
+                    local curNumTradedIn = player:getCharVar(leggingsVarName)
+                    local numTradingInNow = trade:getItemQty(xi.items.RUSTY_LEGGINGS);
+                    local remainderNeeded = utils.clamp(requiredLeggings - curNumTradedIn, 0, requiredLeggings);
+
+                    local clampedTrade = math.min(numTradingInNow, remainderNeeded) -- Only take the amount needed to complete the quest and nothing more
+
+                    if npcUtil.tradeHas(trade, { { xi.items.RUSTY_LEGGINGS, clampedTrade } }) then
+                        local totalWithTrade = curNumTradedIn + clampedTrade
+
+                        if totalWithTrade >= 10 then
+                            return quest:progressEvent(129) -- Complete the quest
+                        elseif totalWithTrade <= 9 then
+                            player:setCharVar(leggingsVarName, totalWithTrade) -- Keep track of traded so far
+                            player:confirmTrade()
+                            return quest:event(128, player:getCharVar(leggingsVarName))
+                        end
+                    else
+                        player:tradeComplete(false) -- There were extra items in the trade
+                        return quest:event(128, player:getCharVar(leggingsVarName))
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [129] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:setCharVar(leggingsVarName, requiredLeggings)
+                        player:confirmTrade()
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_COMPLETED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heizo'] = quest:event(126), -- Standard Conversation
```

Comment:
Not needed if in DefaultActions.lua. Only needed here if dialog changes after quest completed.

Reply:
Ok, good to know, i'll remove this.